src-tauri/src/telemetry_manager.rs

```diff
@@ -130,17 +145,23 @@ impl TelemetryManager {
         self.node_network = network;
     }
 
-    pub async fn initialize(&mut self) -> Result<()> {
+    pub async fn initialize(
+        &mut self,
+        app_state: tauri::State<'_, UniverseAppState>,
```

Comment:
I think it is better if we don't give over the entire state, but only the access token (Arc<RwLock<>>) because that would make it more strict